### PR TITLE
[Cogs Face] Align with latest released version of Face API with  feature of multiple recognition models. 

### PR DESCRIPTION
**Pull request by lebronJ**
(I am one of the main devs working on Face API and Face containers.)

This is to align Face API swaggers **with latest published version** with **Multiple Recognition Model** features. (rollout will be started as soon as we get final approval from API review side)
* We have locally generated AutoRest [.NET SDK](https://github.com/acured/azure-sdk-for-net/pull/4) and written tests to ensure all functionalities work.
* This should also generate the documentation for both [Azure REST API Reference](https://docs.microsoft.com/en-us/rest/api/cognitiveservices/face/face/detectwithurl) and [Face .NET SDK documentation](https://docs.microsoft.com/en-us/dotnet/api/overview/azure/cognitiveservices/client/face?view=azure-dotnet).

Note: This PR has 
1. Added a parameter to existing API interfaces.
2. Added a new field in API response json schema.

I understand this will be a **breaking change** from SDK perspective while at least this will not affect existing SDK users before they upgrade to latest SDK version. We have asked an exception from Azure API Review Board for this.

### Latest improvements:
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i><br>
### Contribution checklist:
- [x] I have reviewed the [documentation](https://github.com/Azure/azure-rest-api-specs#basics) for the workflow.
- [x] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.
- [x] The [OpenAPI Hub](https://aka.ms/openapiportal) was used for checking validation status and next steps.
### ARM API Review Checklist
- [ ] Service team MUST add the "WaitForARMFeedback" label if the management plane API changes fall into one of the below categories. 
- adding/removing APIs.
- adding/removing properties.
- adding/removing API-version. 
- adding a new service in Azure.

Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs.
- [ ] If you are blocked on ARM review and want to get the PR merged urgently, please get the ARM oncall for reviews (RP Manifest Approvers team under Azure Resource Manager service) from IcM and reach out to them. 
Please follow the link to find more details on [API review process](https://armwiki.azurewebsites.net/rp_onboarding/ResourceProviderOnboardingAPIRevieworkflow.html).

### CODE REVIEW

**Comment by lebronJ** [comment]
@johan can help explain more on this and I would appreciate if you could kindly offer us a sign-off from Azure API review's perspective, thanks.

**Comment by shahabhijeet** [comment]
startbuild

**Comment by shahabhijeet** [comment]
startbuild

**Comment by shahabhijeet** [comment]
startbuild

**Comment by shahabhijeet** [comment]
startbuild

**Comment by adxsdknet** [comment]
# Automation for azure-sdk-for-net 
A PR has been created for you: 
https://github.com/azure/azure-sdk-for-net/pull/5457 
.NET SDK Commits: 
https://github.com/adxsdknet/azure-sdk-for-net/pull/5457/commits/f316362680fdf098b98578867f70906d97860a9a

**Comment by veronicagg** [comment]
@lebronJ thanks a lot for the detailed comments at the top, since they answered several of my questions from the start :) the change looks good to me. I will add the potential-breaking-changes label so SDK owners know. Are the API changes deployed by now? If so, I can approve and merge this change.

**Review by veronicagg — CHANGES_REQUESTED** [review]
please see my previous comment.

**Comment by lebronJ** [comment]
> @lebronJ thanks a lot for the detailed comments at the top, since they answered several of my questions from the start :) the change looks good to me. I will add the potential-breaking-changes label so SDK owners know. Are the API changes deployed by now? If so, I can approve and merge this change.

@veronicagg Thanks a lot for your quick review while our asking for exception to Azure API review is still on-going. Please hold on this a little bit and will let you know when we reach agreement on merging this.

File: specification/cognitiveservices/data-plane/Face/stable/v1.0/Face.json [hunk 1]
```diff
@@ -680,7 +680,7 @@
     },
     "/persongroups": {
       "get": {
-        "description": "List person groups and their information.",
+        "description": "List person groups’s pesonGroupId, name, userData and recognitionModel.<br />\n* Person groups are stored in alphabetical order of personGroupId.\n* \"start\" parameter (string, optional) is a user-provided personGroupId value that returned entries have larger ids by string comparison. \"start\" set to empty to indicate return from the first item.\n* \"top\" parameter (int, optional) specifies the number of entries to return. A maximal of 1000 entries can be returned in one call. To fetch more, you can specify \"start\" with the last retuned entry’s Id of the current call.\n<br />\nFor example, total 5 person groups: \"group1\", ..., \"group5\".\n<br /> \"start=&top=\" will return all 5 groups.\n<br /> \"start=&top=2\" will return \"group1\", \"group2\".\n<br /> \"start=group2&top=3\" will return \"group3\", \"group4\", \"group5\".\n",
```
Comment (nschonni) [inline specification/cognitiveservices/data-plane/Face/stable/v1.0/Face.json]: pesonGroupId -> personGroupId

File: specification/cognitiveservices/data-plane/Face/stable/v1.0/Face.json [hunk 2]
```diff
@@ -1579,7 +1582,7 @@
     },
     "/largepersongroups": {
       "get": {
-        "description": "List large person groups and their information.",
+        "description": "List all existing large person groups’s largePesonGroupId, name, userData and recognitionModel.<br />\n* Large person groups are stored in alphabetical order of largePersonGroupId.\n* \"start\" parameter (string, optional) is a user-provided largePersonGroupId value that returned entries have larger ids by string comparison. \"start\" set to empty to indicate return from the first item.\n* \"top\" parameter (int, optional) specifies the number of entries to return. A maximal of 1000 entries can be returned in one call. To fetch more, you can specify \"start\" with the last retuned entry’s Id of the current call.\n<br />\nFor example, total 5 large person groups: \"group1\", ..., \"group5\".\n<br /> \"start=&top=\" will return all 5 groups.\n<br /> \"start=&top=2\" will return \"group1\", \"group2\".\n<br /> \"start=group2&top=3\" will return \"group3\", \"group4\", \"group5\".\n",
```
Comment (nschonni) [inline specification/cognitiveservices/data-plane/Face/stable/v1.0/Face.json]: largePesonGroupId-> largePersonGroupId

Comment (lebronJ) [reply specification/cognitiveservices/data-plane/Face/stable/v1.0/Face.json]: Done, thanks.

File: specification/cognitiveservices/data-plane/Face/stable/v1.0/Face.json (hunk 1 above)
Comment (lebronJ) [reply specification/cognitiveservices/data-plane/Face/stable/v1.0/Face.json]: Done, thanks.

**Comment by yangyuan** [comment]
This involves API design changes. Have you finished the API design review?

**Comment by lebronJ** [comment]
Hi @yangyuan , could you please first add a Do-Not-Merge label for this PR?

Actually we are now asking the API review board for exception and we are also applying some API interface changes for service side now upon the review comments. There will be changes for the Swagger after we finalize the implementation for service side and get review sign-off.

Will ping back this PR then.

**Comment by veronicagg** [comment]
@lebronJ would you prefer to close this PR and re-open when you're ready? you can tag me back when ready. Thanks!

File: specification/cognitiveservices/data-plane/Face/stable/v1.0/Face.json [hunk 3]
```diff
@@ -4199,6 +4230,16 @@
         "recognition_01",
         "recognition_02"
       ]
+    },    
+    "returnRecognitionModel": {
+      "name": "returnRecognitionModel",
+      "description": "Whether to return the 'RecognitionModel' required for the current operation.",
```
Comment (lebronJ) [inline specification/cognitiveservices/data-plane/Face/stable/v1.0/Face.json]: `"description": "A value indicating whether the operation should return 'recognitionModel' in response.",`

Comment (acured) [reply specification/cognitiveservices/data-plane/Face/stable/v1.0/Face.json]: Changed.

**Comment by lebronJ** [comment]
> @lebronJ would you prefer to close this PR and re-open when you're ready? you can tag me back when ready. Thanks!

Hi @veronicagg and @yangyuan, we have gotten the API review sign-off and I have forwarded it to Cognitive Services Experience Team and you two. Can we move forward to review and merge this as soon as possible (thanks), so that we can catch the dates of our roll out, along with the API reference updates, SDKs and samples?

**Comment by lebronJ** [comment]
> > @lebronJ would you prefer to close this PR and re-open when you're ready? you can tag me back when ready. Thanks!
> 
> Hi @veronicagg and @yangyuan, we have gotten the API review sign-off and I have forwarded it to Cognitive Services Experience Team and you two. Can we move forward to review and merge this as soon as possible (thanks), so that we can catch the dates of our roll out, along with the API reference updates, SDKs and samples?

Besides, could anyone please help remove the DoNotMerge label and reopen the PR of SDK changes? Let me know if any action needed from my side.

**Review by yangyuan — APPROVED** [review]

**Review by veronicagg — APPROVED** [review]
Based on the above, the specs submitted reflect your APIs which have been deployed, so I'm approving and merging. To request SDKs releases, please use https://portal.azure-devex-tools.com/app/tools/request-api-release . Thanks!

**Comment by jhendrixMSFT** [comment]
@AutorestCI regenerate azure-sdk-for-go

**Pull request merged**